tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -537,18 +1070,146 @@ public void testPartialIteratingTryWithResource() throws Exception {
         getRootPath(); // will fail on windows if iterator not closed correctly
     }
 
+    @Test(expected = RealmUnavailableException.class)
+    public void testMismatchSecretKey() throws Exception {
```

Comment:
Do you have a similar test for integrity (i.e., wrong private key / public key)

Reply:
Just added `testMismatchPublicKey()` and `testMismatchPrivateKey()`